### PR TITLE
Prevent basic auth from blocking robots.txt

### PR DESCRIPTION
**Pull request by robindevitt**
Basic auth blocks access to robots.txt which is generated by WP.

This PR allows access to robots.txt when basic auth is enabled.

This PR also sets the robots.txt file to return:
```
User-agent: *
Disallow: /
```

### Steps to confirm:
1. Enable `PHP Basic Auth` on your environment.
2. Confirm `PHP Basic Auth` is active by trying to view a page on the site.
3. Once enabled, try access the `/robots.txt` file and you will see the below:
```
User-agent: *
Disallow: /
```

[Original issue #154](https://github.com/humanmade/altis-security/issues/154)

### CODE REVIEW

**Comment by robindevitt** [comment]
@shadyvb when you have a moment, please could you have a look at these two Travis CI Build errors ?

File: inc/namespace.php [hunk 1]
```diff
@@ -150,6 +150,15 @@ function require_auth() {
 		return;
 	}
 
+	$allowed = [
+		'/robots.txt',
+	];
+
+	if ( $_SERVER['REQUEST_URI'] && in_array( $_SERVER['REQUEST_URI'], $allowed, true ) ) {
```
Comment (shadyvb) [inline inc/namespace.php:162]: Same as the other PR, this condition is only true for this specific URL, so having an `$allowed` array gives the impression it applies to more cases. Let's stick with a simple string comparison here.

**Review by shadyvb — CHANGES_REQUESTED** [review]

File: inc/namespace.php (hunk 1 above)
Comment (rmccue) [reply inc/namespace.php:162]: Could we leave it as an array but make it filterable? It's useful in other cases, and we have [an equivalent for hm-require-login](https://github.com/humanmade/hm-require-login/blob/master/inc/namespace.php#L44) (which we use for allowing eg OAuth through)

**Review by shadyvb — CHANGES_REQUESTED** [review]
LGTM, need to tackle the linter failures and should be good to go.

**Comment by robindevitt** [comment]
@shadyvb I have the HM Linter changes sorted, but having the Travis CI builds error. Going to try have a look at it today otherwise tomorrow.

**Review by shadyvb — APPROVED** [review]

**Pull request merged**